agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go

```diff
@@ -519,9 +515,34 @@ func (fv *FSxWindowsFileServerResource) retrieveFileSystemDNSName(fileSystemId s
 	return nil
 }
 
+var mountLock sync.Mutex
```

Comment:
Why do you need an additional lock while performing `hostMount`? Please add a comment to explain the rationale. Also, why is this outside the resource encapsulation?

Reply:
I'll add a comment for this